client/src/app/header/search-typeahead.component.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  Component,
+  ViewChild,
+  ElementRef,
+  AfterViewInit,
+  OnInit,
+  OnDestroy,
+  QueryList
+} from '@angular/core'
+import { Router, NavigationEnd, Params, ActivatedRoute } from '@angular/router'
+import { AuthService, ServerService } from '@app/core'
+import { I18n } from '@ngx-translate/i18n-polyfill'
+import { filter, first, tap, map } from 'rxjs/operators'
+import { ListKeyManager } from '@angular/cdk/a11y'
+import { UP_ARROW, DOWN_ARROW, ENTER } from '@angular/cdk/keycodes'
+import { SuggestionComponent, Result } from './suggestion.component'
+import { of } from 'rxjs'
+import { getParameterByName } from '@app/shared/misc/utils'
+import { ServerConfig } from '@shared/models'
+
+@Component({
+  selector: 'my-search-typeahead',
+  templateUrl: './search-typeahead.component.html',
+  styleUrls: [ './search-typeahead.component.scss' ]
+})
+export class SearchTypeaheadComponent implements OnInit, OnDestroy, AfterViewInit {
+  @ViewChild('contentWrapper', { static: true }) contentWrapper: ElementRef
+
+  hasChannel = false
+  inChannel = false
+  newSearch = true
+
+  searchInput: HTMLInputElement
+  serverConfig: ServerConfig
+
+  URIPolicyText: string
+  inAllText: string
+  inThisChannelText: string
+  globalSearchIndex = 'https://index.joinpeertube.org'
+
+  keyboardEventsManager: ListKeyManager<SuggestionComponent>
+  results: any[] = []
+
+  constructor (
+    private authService: AuthService,
+    private router: Router,
+    private route: ActivatedRoute,
+    private serverService: ServerService,
+    private i18n: I18n
+  ) {
+    this.URIPolicyText = this.i18n('Determines whether you can resolve any distant content, or if your instance only allows doing so for instances it follows.')
+    this.inAllText = this.i18n('In all PeerTube')
+    this.inThisChannelText = this.i18n('In this channel')
+  }
+
+  ngOnInit () {
+    this.router.events
+      .pipe(filter(e => e instanceof NavigationEnd))
+      .subscribe((event: NavigationEnd) => {
+        this.hasChannel = event.url.startsWith('/videos/watch')
+        this.inChannel = event.url.startsWith('/video-channels')
+        this.computeResults()
+      })
+
+    this.router.events
+      .pipe(
+        filter(e => e instanceof NavigationEnd),
+        map(() => getParameterByName('search', window.location.href))
+      )
+      .subscribe(searchQuery => this.searchInput.value = searchQuery || '')
```

Comment:
Do you really want to subscribe to a search change? Or is it just to init the search input on first load? Because it could lead to a loop: `url changed` -> `update the input` -> `input changed` -> `update the url` -> `url changed` etc

Reply:
indeed. corrected.